src/xenia/kernel/xevent.cc

```diff
@@ -70,5 +71,33 @@ int32_t XEvent::Reset() {
 
 void XEvent::Clear() { event_->Reset(); }
 
+bool XEvent::Save(ByteStream* stream) {
+  XELOGD("XEvent %.8X (%s)", handle(), manual_reset_ ? "manual" : "auto");
+  SaveObject(stream);
+
+  // TODO: Write if we're currently signaled.
```

Comment:
you can wait for the event with a zero timeout and check the return code - if it is `WAIT_TIMEOUT` you know it wasn't signaled.